res/controllers/Midi_for_light-scripts.js

```diff
@@ -16,33 +16,33 @@ function midi_for_light() {}
 ///////////////////////////////////////////////////////////////
 //                       USER OPTIONS                        //
 ///////////////////////////////////////////////////////////////
-var midi_channel = 1; // set midi_channel. Valid range: 1 to 16.
-var enable_beat = true; // set to false if you not need beat
-var enable_bpm = true; // set to false if you not need BPM
-var enable_mtc_timecode = false; // set to false if you not need midi time code
-var enable_vu_mono_current = false; // set to false if you not need VU mono current
-var enable_vu_mono_average_min = false; // set to false if you not need VU mono average min
-var enable_vu_mono_average_mid = false; // set to false if you not need VU mono average mid
-var enable_vu_mono_average_max = false; // set to false if you not need VU mono average max
-var enable_vu_mono_average_fit = true; // set to false if you not need VU mono average fit
-var enable_vu_mono_current_meter = false; // set to false if you not need VU mono current meter
-var enable_vu_mono_average_meter = true; // set to false if you not need VU mono average meter
-var enable_vu_left_current = false; // set to false if you not need VU left current
-var enable_vu_left_average_min = false; // set to false if you not need VU left average min
-var enable_vu_left_average_mid = false; // set to false if you not need VU left average mid
-var enable_vu_left_average_max = false; // set to false if you not need VU left average max
-var enable_vu_left_average_fit = true; // set to false if you not need VU left average fit
-var enable_vu_left_current_meter = false; // set to false if you not need VU left current meter
-var enable_vu_left_average_meter = false; // set to false if you not need VU left average meter
-var enable_vu_right_current = false; // set to false if you not need VU right current
-var enable_vu_right_average_min = false; // set to false if you not need VU right average min
-var enable_vu_right_average_mid = false; // set to false if you not need VU right average mid
-var enable_vu_right_average_max = false; // set to false if you not need VU right average max
-var enable_vu_right_average_fit = true; // set to false if you not need VU right average fit
-var enable_vu_right_current_meter = false; // set to false if you not need VU right current meter
-var enable_vu_right_average_meter = false; // set to false if you not need VU right average meter
-var deck_ending_time = 15; // set a time (in seconds) in which the playing track is considered to be ending
-var deck_ending_priority_factor = 0.9; // decrease the priority of the ending track by this factor
+var midi_channel = engine.getSetting("midi_channel"); // set midi_channel. Valid range: 1 to 16.
```

Comment:
In previous mappings we've "defensively" added the default values here too (either using the nullish coalescing operator `??` or `||`). I don't think that should be necessary, so this looks good to me, but just to confirm, is this safe @acolombier?

Reply:
AFAIK, the default values there were only necessarry for backward compatibility of the mapping to 2.4 without the controller settings feature.

Reply:
Ah interesting, so `engine.getSetting` was already available in 2.4, but invocations always returned `undefined`?

Reply:
It does not exist in 2.4, and is therefore `undefined`.

Reply:
You can't call `undefined` without getting a `TypeError` though.

Reply:
I can confirm the mapping does not work with the proposed fallback. Changing the line to
`var midi_channel = engine.getSetting("midi_channel") || 1;`
causes an error in Version 2.4:
`Uncaught exception: file:///C:/Program Files/Mixxx/controllers/Midi_for_light-scripts.js:19: TypeError: Property 'getSetting' of object ControllerScriptInterfaceLegacy(0x16b9d336520) is not a function`

Reply:
`var midi_channel = engine.getSetting ? engine.getSetting("midi_channel") : 1;` seems to work

Reply:
Depending on how new the JS engine is you might be able to do

```ts
engine.getSetting?.("midi_channel") ?? 1
```

But tbh if we're merging this to 2.5, it shouldn't be needed anyway (and I don't think there'll be many 2.4 releases anymore apart from minor bug fixes)